openhands/tools/execute_bash/definition.py

```diff
@@ -99,7 +99,15 @@ def agent_observation(self) -> Sequence[TextContent | ImageContent]:
             ret += f"\n[Command finished with exit code {self.metadata.exit_code}]"
         if self.error:
             ret = f"[There was an error during command execution.]\n{ret}"
-        return [TextContent(text=maybe_truncate(ret, MAX_CMD_OUTPUT_SIZE))]
+
+        # Use enhanced truncation with file saving if working directory is available
+        truncated_text = maybe_truncate(
+            content=ret,
+            truncate_after=MAX_CMD_OUTPUT_SIZE,
+            save_dir=self.metadata.working_dir,
```

Comment:
ditto: We should accept a `save_dir` from BashTool set .create() instead of cwd so it doesn't get into VCS.

Maybe we can default it to `/tmp/.openhands`